### ISSUE TITLE
Unique constraint violation on unique functional index with nullable fields

### ISSUE THREAD

**Issue by RojinBaka** [tarantool/tarantool#8587]
**Bug description**

Tarantool allows insertion of records that violate 'unique' constraint on functional index if index parts are marked as nullable (changing fields to non-nullable produces expected result)

Tarantool 2.10.6-0-g3990f976b (latest docker image)

**Steps to reproduce**

```lua
#!/usr/bin/env tarantool
box.cfg{}
s = box.schema.space.create('withdata')
s:format({
	{name = 'name', type = 'string'},
    {name = 'address', type = 'string'},
    {name = 'address2', type = 'string'}
	})
pk = s:create_index('name', {parts = {{field = 1, type = 'string'}}})
lua_code = [[function(tuple)
               local address = string.split(tuple[2])
               local ret = {}
               for _, v in pairs(address) do
                 table.insert(ret, {utf8.upper(v), tuple[3]})
               end
               return ret
             end]]
box.schema.func.create('address',
                        {body = lua_code,
                         is_deterministic = true,
                         is_sandboxed = true,
                         opts= {is_multikey = true}})
idx = s:create_index('addr', {unique = true,
                              func = 'address',
                              parts = {
					{field = 1, type = 'string', is_nullable = true},
					{field = 2, type = 'string', is_nullable = true}
                              }})
s:insert({"James", "SIS Building Lambeth London UK", "1"})
s:insert({"Sherlock", "221B Baker St Marylebone London NW1 6XE UK", "2"})
s:insert({"Sherlock2", "221B Baker St Marylebone London NW1 6XE UK", "2"})
return idx:select('UK')
```

**Actual behavior**

3 records are inserted (2nd and 3rd having duplicate index values), 
```lua
tarantool> s:insert({"Sherlock2", "221B Baker St Marylebone London NW1 6XE UK", "2"})
---
- ['Sherlock2', '221B Baker St Marylebone London NW1 6XE UK', '2']
...

tarantool> return idx:select('UK')
---
- - ['James', 'SIS Building Lambeth London UK', '1']
  - ['Sherlock', '221B Baker St Marylebone London NW1 6XE UK', '2']
  - ['Sherlock2', '221B Baker St Marylebone London NW1 6XE UK', '2']

```
**Expected behavior**

3-rd insert should fail with error (as 2nd and 3rd records have same values for unique index 'addr'), and only two records should be present in space
```lua
tarantool> s:insert({"Sherlock2", "221B Baker St Marylebone London NW1 6XE UK", "2"})
---
- error: Duplicate key exists in unique index "addr" in space "withdata" with old
    tuple - ["Sherlock", "221B Baker St Marylebone London NW1 6XE UK", "2"] and new
    tuple - ["Sherlock2", "221B Baker St Marylebone London NW1 6XE UK", "2"]

tarantool> return idx:select('UK')
---
- - ['James', 'SIS Building Lambeth London UK', '1']
  - ['Sherlock', '221B Baker St Marylebone London NW1 6XE UK', '2']
```

**Comment by Gumix** [comment]
A bit simplified reproducer:
```lua
box.cfg{}
s = box.schema.space.create('test1')
s:format({{name = 'id', type = 'unsigned'},
          {name = 'c2', type = 'string'}})
s:create_index('pk')

box.schema.func.create('my_func',
                        {is_sandboxed = true,
                         is_deterministic = true,
                         body = "function(tuple) return {tuple[2]} end"})

idx = s:create_index('sk', {func = 'my_func',
                            unique = true,
                            parts = {{field = 1,
                                      is_nullable = true,
                                      type = 'string'}}})
s:insert{1, 'aa'}
s:insert{2, 'aa'}
idx:select('aa')
```
Actual behavior:
```
---
- - [1, 'aa']
  - [2, 'aa']
...
```
Without `is_nullable = true `:
```
---
- error: Duplicate key exists in unique index "sk" in space "test1" with old tuple
    - [1, "aa"] and new tuple - [2, "aa"]
...
```

**Comment by locker** [comment]
Difficult to cherry-pick to 2.10. Discussed with @alyapunov. Since the bug isn't critical, decided to cherry-pick the fix only to 2.11.
